mozilla-services/consvc-shepherd
# [SHEP-15] when new deal is imported into shepherd automatically create one child campaign for that deal

## References

[SHEP-15](https://mozilla-hub.atlassian.net/browse/SHEP-15)

## Problem Statement

Now that we have the campaigns schema, we need to create a new child campaign for the newly created deal when the Boostr script is executed and the deals are imported.

## Proposed Changes
1. Updated the Boostr sync script to create campaigns only for newly created deals.
2. Added a mock for the `create_campaign` function and ensured it is called when the Boostr script tests are executed.
3. Modified the Campaigns model to make `ad_ops_person`, `notes`, and `kevel_flight_id` nullable, as these fields will be manually populated by Ad Ops after the child campaign is created.

## Verification Steps
1. Execute the command to open the Shepherd app's container shell: `docker exec -it consvc-shepherd-app-1 sh`
2. Run the Boostr script:` python manage.py sync_boostr_data https://app.boostr.com/api/`
3. Check if new campaigns are created for the newly imported deals.



[SHEP-15]: https://mozilla-hub.atlassian.net/browse/SHEP-15?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
